ahnlich/db/src/engine/predicate.rs

```diff
@@ -255,25 +255,32 @@ impl PredicateIndex {
     }
     /// checks the predicate index for a predicate op and value. The return type is a StdHashSet<_>
     /// because we do not modify it at any point so we do not need concurrency protection
-    fn matches<'a>(
-        &self,
-        predicate_op: &'a PredicateOp,
-        value: &'a MetadataValue,
-    ) -> StdHashSet<StoreKeyId> {
+    fn matches(&self, predicate: &Predicate) -> StdHashSet<StoreKeyId> {
         let pinned = self.0.pin();
-        match predicate_op {
-            PredicateOp::Equals => {
+
+        match predicate {
+            Predicate::Equals { value, .. } => {
```

Comment:
Probably a Nit Pick or a question, but isn't this suppose to be the key?

Reply:
Nah its the value you compare against